src/main/java/core/basesyntax/service/impl/StorageServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.StorageService;
+
+public class StorageServiceImpl implements StorageService {
+    @Override
+    public void decrement(Fruit fruit, int quantity) {
+        int currentQuantity = Storage.fruitInventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new IllegalArgumentException("Not enough " + fruit + " in storage");
```

Comment:
Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Here, instead of `IllegalArgumentException`, throw a `RuntimeException` with a more informative message to adhere to the checklist.